crates/wasi-provider/src/lib.rs

```diff
@@ -18,22 +18,35 @@ use handle::PodHandle;
 use wasi_runtime::WasiRuntime;
 
 const TARGET_WASM32_WASI: &str = "wasm32-wasi";
+const LOG_DIR_NAME: &str = "wasi-logs";
 
 /// WasiProvider provides a Kubelet runtime implementation that executes WASM
 /// binaries conforming to the WASI spec
 #[derive(Clone)]
 pub struct WasiProvider {
     handles: Arc<RwLock<HashMap<String, PodHandle<File>>>>,
     store: FileModuleStore,
+    log_path: PathBuf,
 }
 
 impl WasiProvider {
-    pub fn new() -> Self {
-        let store = FileModuleStore::new(&dirs::home_dir().expect("Cannot get home directory"));
-        Self {
+    /// Returns a new WASI provider configured to use the proper data directory
+    /// (including creating it if necessary)
+    pub fn new<P: AsRef<Path>>(data_dir: P) -> anyhow::Result<Self> {
+        // Make sure we have a log dir and containers dir created
+        let data_path = data_dir.as_ref().to_path_buf();
+        // This is temporary as we should probably be passing in a ModuleStore
+        // as a parameter or the oci client as a whole
+        // NOTE: We do not have to create the dir here as the FileModuleStore already does this
+        let container_path = data_path.join("containers");
+        let log_path = data_path.join(LOG_DIR_NAME);
+        std::fs::create_dir_all(&log_path)?;
```

Comment:
We should use `tokio::fs::create_dir_all` and make the function `async`

Reply:
I actually did that first and it made no sense as the provider should be created before passing it in anyway.

However, with that said, because `main` is async, is it safer to just do a Tokyo call here?

Reply:
Basically anything that's being executed in an async context should be an async call.

Reply:
I guess the other consideration here is around whether or not users should have to be in an async block to create this provider and my initial feeling is no

Reply:
But we use it in an async block currently so at that site when we need to call spawn_blocking